### PR TITLE
[WOE] Implement Agatha's Soul Cauldron

### PR DESCRIPTION
**Pull request by theelk801**
the test should be enough coverage

### CODE REVIEW

**Comment by xenohedron** [comment]
(cancelling the PR build since there was a broken commit, the branch build should be fine)

File: Mage.Sets/src/mage/cards/a/AgathasSoulCauldron.java [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import mage.target.common.TargetCardInGraveyard;
+import mage.target.common.TargetControlledCreaturePermanent;
+import mage.util.CardUtil;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static mage.constants.Outcome.Benefit;
```
Comment (xenohedron) [inline Mage.Sets/src/mage/cards/a/AgathasSoulCauldron.java:36]: I'm not a fan of static imports, don't think it adds value here

**Review by xenohedron — APPROVED** [review]
looks good to me

File: Mage.Sets/src/mage/cards/a/AgathasSoulCauldron.java (hunk 1 above)
Comment (theelk801) [reply Mage.Sets/src/mage/cards/a/AgathasSoulCauldron.java]: oh that shouldn't have been there, thanks for catching

**Pull request merged**